Sources/Tools/MaxMain/MaxCompat.h

```diff
@@ -175,4 +185,121 @@ static_assert(std::is_same_v<MCHAR, TCHAR>, "MCHAR and TCHAR must have the same
 #   endif
 #endif
 
+// Overload proxies
+template<class T>
+class plMaxClassDesc : public T
+{
+    static_assert(std::is_base_of_v<class ClassDesc, T>, "plMaxClassDesc can only be used for ClassDesc subclasses");
+
+public:
+#if MAX_VERSION_MAJOR >= 24 // Max 2022
+    const MCHAR* NonLocalizedClassName() override { return ClassName(); }
+#endif
+};
+
+template<class T>
+class plMaxAnimatable : public T
+{
+    static_assert(std::is_base_of_v<class Animatable, T>, "plMaxAnimatable can only be used for Animatable subclasses");
+
+protected:
+    // Sealed in Max 2022
+    virtual void IGetClassName(MSTR& s) const { s = _M("Animatable"); }
+    virtual MSTR ISubAnimName(int i) { return _M("no name"); }
+
+public:
+#if MAX_VERSION_MAJOR < 24
+    void GetClassName(MSTR& s) MAX24_CONST override { IGetClassName(s); }
+    void GetClassName(MSTR& s, bool localized) MAX24_CONST { IGetClassName(s); }
+    MSTR SubAnimName(int i) override { return ISubAnimName(i); }
+    MSTR SubAnimName(int i, bool localized) { return ISubAnimName(i); }
+#else
+    void GetClassName(MSTR& s, bool localized) MAX24_CONST override { return IGetClassName(s); }
```

Comment:
This is already in `#else` from `#if MAX_VERSION_MAJOR < 24`
```suggestion
    void GetClassName(MSTR& s, bool localized) const override { return IGetClassName(s); }
```